src/platform/packages/shared/kbn-scout/src/execution/test_track.test.ts

```diff
@@ -0,0 +1,546 @@
+  it('should throw an error if no lanes are available and allowNewLane is false', () => {
+    const track = new TestTrack({ runtimeTarget: 10 });
+    expect(track.laneCount).toBe(0);
+
+    const load: TestTrackLoad = {
+      id: 'configPathA',
+      stats: {
+        runCount: 0,
+        runtime: {
+          avg: 0,
+          median: 0,
+          pc95th: 7,
+          pc99th: 0,
+          max: 0,
+          estimate: 7,
+        },
+      },
+    };
+
+    expect(() => track.addLoadToLeastCongestedLane(load, false)).toThrowError(
+      "Track doesn't have any lanes"
+    );
+  });
+
+  it('returns the test track specification', () => {
+    const track = new TestTrack({ runtimeTarget: 10 });
+
+    const loadA: TestTrackLoad = {
+      id: 'configPathA',
+      stats: {
+        runCount: 0,
+        runtime: {
+          avg: 0,
+          median: 0,
+          pc95th: 7,
+          pc99th: 0,
+          max: 0,
+          estimate: 7,
+        },
+      },
+    };
+
+    track.addLoadToLeastCongestedLane(loadA, true);
+
+    const loadB: TestTrackLoad = {
+      id: 'configPathB',
+      stats: {
+        runCount: 0,
+        runtime: {
+          avg: 0,
+          median: 0,
+          pc95th: 8,
+          pc99th: 0,
+          max: 0,
+          estimate: 8,
+        },
+      },
+    };
+
+    track.addLoadToLeastCongestedLane(loadB, true);
+
+    // adding a third load will add the load to the least loaded lane: the first lane
+    const loadC: TestTrackLoad = {
+      id: 'configPathC',
+      stats: {
+        runCount: 0,
+        runtime: {
+          avg: 0,
+          median: 0,
+          pc95th: 2,
+          pc99th: 0,
+          max: 0,
+          estimate: 2,
+        },
+      },
+    };
+
+    // disable creating a new lane
+    track.addLoadToLeastCongestedLane(loadC, false);
+
+    const loadD: TestTrackLoad = {
+      id: 'configPathD',
+      stats: {
+        runCount: 0,
+        runtime: {
+          avg: 0,
+          median: 0,
+          pc95th: 5,
+          pc99th: 0,
+          max: 0,
+          estimate: 5,
+        },
+      },
+    };
+
+    track.addLoadToLeastCongestedLane(loadD, false);
+
+    expect(track.specification).toEqual({
+      lanes: [
+        {
+          availableCapacity: 1,
+          isCongested: false,
+          loads: ['configPathA', 'configPathC'],
+          number: 1,
+          runtimeEstimate: 9,
+          runtimeTarget: 10,
+          status: 'open',
+        },
+        {
+          availableCapacity: -3,
+          isCongested: true,
+          loads: ['configPathB', 'configPathD'],
+          number: 2,
+          runtimeEstimate: 13,
+          runtimeTarget: 10,
+          status: 'closed',
+        },
+      ],
+      stats: {
+        combinedRuntime: {
+          expected: 22,
+          overflow: 2,
+          target: 20,
+          unused: 0,
+        },
+        lane: {
+          count: 2,
+          longestEstimate: 13,
+          saturationPercent: 110,
```

Comment:
Is it normal for the saturation percent to go over 110 in this example?

Reply:
Yep. It's an indicator for overloading/congestion.